src/test/java/kitchenpos/domain/OrderTableTest.java

```diff
@@ -0,0 +1,105 @@
+package kitchenpos.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import kitchenpos.fixtures.OrderFixtures;
+import kitchenpos.fixtures.TableFixtures;
+import org.junit.jupiter.api.Test;
+
+class OrderTableTest {
+
+    @Test
+    void 그룹을_지정한다() {
+        OrderTable orderTable = TableFixtures.createOrderTable(true);
+        TableGroup tableGroup = TableFixtures.createTableGroup();
+        orderTable.belongsTo(tableGroup);
+
+        assertThat(orderTable.getTableGroup()).isNotNull();
+    }
+
+    @Test
+    void 그룹_지정_시_비어있지_않으면_예외를_반환한다() {
+        OrderTable notEmptyTable = TableFixtures.createOrderTable(false);
+        TableGroup tableGroup = TableFixtures.createTableGroup();
+
+        Exception exception = assertThrows(IllegalStateException.class, () -> notEmptyTable.belongsTo(tableGroup));
+        assertThat(exception.getMessage()).isEqualTo("빈 테이블이 아닙니다.");
+    }
+
+    @Test
+    void 그룹_지정_시_이미_지정되어_있으면_예외를_반환한다() {
+        OrderTable groupedTable = TableFixtures.createGroupedOrderTable(true);
+        TableGroup tableGroup = TableFixtures.createTableGroup();
+
+        Exception exception = assertThrows(IllegalStateException.class, () -> groupedTable.belongsTo(tableGroup));
+        assertThat(exception.getMessage()).isEqualTo("이미 단체 지정된 테이블입니다.");
```

Comment:
```suggestion
        assertThatCode(() -> groupedTable.belongsTo(tableGroup))
                .isInstanceOf(IllegalStateException.class)
                .hasMessage("이미 단체 지정된 테이블입니다.");
```

이렇게 하는 것도 가능해요. `assertThrows()`는 Junit의 메서드고 `assertThatCode()`는 AssertJ의 메서드에요. 
`assertThatCode()`이후에 다양한 메서드를 쓸 수 있어서 저는 훨씬 테스트하기 편한 것 같아요. `isInstanceOf()`말고 `isExactlyInstanceOf()`를 쓰면 예외의 정확한 클래스를 검증할 수 있어서 활용도가 높아요. 

`assertThrows()` & `assertThatCode().isInstanceOf()` 는 `RuntimeException.class`이 발생하는지 테스트할 때 자식 클래스인 `IllegalStateException`이 발생해도 통과돼요. 내부적으로 instanceof 연산자를 쓰고 있거든요.

반면 `assertThatCode().isExactlyInstanceOf()` 는 테스트하려는 예외클래스가 아니라면 자식클래스여도 실패해요. 내부에는 `발생하길_기대하는_예외클래스.equals(예외객체.getClass())` 이런식으로 구현되어 있습니다.

Reply:
오 assertThatCode는 몰랐는데 감사합니다! 
isExactlyInstanceOf 때문에라도 앞으로는 assertThatCode 혹은 assertThatThrownBy를 사용할 것 같네요. 
하지만 지금은 상위의 예외 클래스를 검증하는 부분은 없어 당장은 적용하진 않을게요 ㅎㅎ